tests/torchtune/data/test_converters.py

```diff
@@ -0,0 +1,120 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+from torchtune.data import openai_to_llama2_messages, sharegpt_to_llama2_messages
+from torchtune.data._types import Message
+
+# Taken from Open-Orca/SlimOrca-Dedup on Hugging Face:
+# https://huggingface.co/datasets/Open-Orca/SlimOrca-Dedup
+CHAT_SAMPLE = {
+    "system": "You are an AI assistant. User will you give you a task. Your goal is to complete the task as faithfully as you can. While performing the task think step-by-step and justify your steps.",  # noqa: B950
+    "user": "Please briefly summarize this news article:\n\nAOL.com Video - Father Lets 8-Year-Old Drive On Icy Road\n\nDescription:Would you let your 8-year-old drive your car? How about on an icy road? Well one father in Russia did just that, and recorded the entire thing. To her credit, the child seemed to be doing a great job. (0:44)\n\nTags: 8-year-old driver , caught on camera , child driver , pix11\n\nSummary:",  # noqa: B950
+    "assistant": "A father in Russia allowed his 8-year-old child to drive his car on an icy road and recorded the event. The child appeared to be handling the situation well, showcasing their driving skills despite the challenging conditions.",  # noqa: B950
+}
+
+EXPECTED_MESSAGE = [
+    Message(
+        role="system",
+        content=CHAT_SAMPLE["system"],
+    ),
+    Message(
+        role="user",
+        content=CHAT_SAMPLE["user"],
+    ),
+    Message(
+        role="assistant",
+        content=CHAT_SAMPLE["assistant"],
+    ),
+]
+
+
+class TestSharegptToLlama2Messages:
+    samples = [
+        {
+            "from": "system",
+            "value": CHAT_SAMPLE["system"],
+        },
+        {
+            "from": "user",
+            "value": CHAT_SAMPLE["user"],
+        },
+        {
+            "from": "assistant",
+            "value": CHAT_SAMPLE["assistant"],
+        },
+    ]
+
+    def test_conversion(self):
+        converted_messages = sharegpt_to_llama2_messages(self.samples)
+        for converted, expected in zip(converted_messages, EXPECTED_MESSAGE):
+            assert converted == expected
```

Comment:
I'm not sure if equality between `Message` classes will work here, you can either check each field directly, add a `to_dict` method to `Message` and compare the dictionaries, or implement an `__eq__` method for `Message`. I think the `to_dict` method is probably easiest.

Reply:
yes there's a bunch of issue with this test - I'll fix it in the next commit but there's something wrong with my local pre-commit that I'm still fixing :(